.github/workflows/build_and_test_upstream_ci_image.yml

```diff
@@ -0,0 +1,38 @@
+name: "Build and test upstream CI image"
+
+# Controls when the action will run. 
+on:
+  # Triggers the workflow on push or pull request events but only for the master branch
+  pull_request:
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  # This workflow contains a single job called "build"
+  build:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-latest
+    
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - uses: actions/checkout@v2
+
+      # Runs a set of commands using the runners shell
+      - name: "Build the upstream_ci_image inside Dockerfiles"
+        run: |
+          podman --version
+          podman build -t upstream_ci_image:latest -f Dockerfiles/ci/Dockerfile --build-arg OPERATOR_SDK_VERSION=v1.4.0
+      - name: "Run example operator-metadata through upstream_ci_image --> negative test"
+        continue-on-error: true
+        run: |
+          mkdir test_operator_work_dir output_logs
+          podman run -it -v $PWD:/project/operator-test-playbooks -v ./Dockerfiles/ci/example-metadata-without-alm-annotations:/project/operator_dir -v ./output_logs:/project/output -v ./test_operator_work_dir:/project/test_operator_work_dir -e TEST_NAME=prepare_operator_metadata upstream_ci_image:latest
+
+      - name: "Fail when previous step does not fail"
+        run: |
+          grep -inr "Failed due to CSV.metadata.annotations\['alm-examples'\] is set to empty" ./output_logs
+          ls output_logs
+          test -f ./output_logs/parse_operator_metadata_results.json
+          test -f ./output_logs/parsed_operator_data.yml
```

Comment:
How will I run these assertions locally as part of development?

Reply:
These assertions will check for the file's existence and fail if they do not exist.